pkg/tfc_trigger/tfc_trigger.go

```diff
@@ -639,6 +639,8 @@ func (t *TFCTrigger) publishRunToStream(ctx context.Context, run *tfe.Run) error
 		DiscussionID:                         t.GetMergeRequestDiscussionID(),
 		RootNoteID:                           t.GetMergeRequestRootNoteID(),
 		VcsProvider:                          t.GetVcsProvider(),
+		//set Auto Merge if both conditions are met.
+		AutoMerge: cfgWS.AutoMerge && cfgWS.Mode == "apply-before-merge",
```

Comment:
do we want to surface some kind of error message if auto merge is configured for "merge-before-apply" or "tfc-vcs-repo"?

Reply:
oh, similar question for "automerge=true" but global auto merge is disabled.

Reply:
We now log if either condition occurs. Plus the help text in the MR will indicate that a manual merge is required.